serialization/chrome_pak.ksy

```diff
@@ -0,0 +1,90 @@
+meta:
+  id: chrome_pak
+  title: Chrome PAK serialization format
+  file-extension: pak
+  license: CC0-1.0
+  endian: le
+doc: |
+  Format mostly used by Google Chrome and various Android apps to store
+  resources such as translated strings, help messages and images.
+doc-ref:
+  - http://dev.chromium.org/developers/design-documents/linuxresourcesandlocalizedstrings # version 4
+  - https://chromium.googlesource.com/chromium/src/tools/grit/+/22f7a68bb5ad68fe4192d0f34466049038735b9c/grit/format/data_pack.py # version 4
+  - https://chromium.googlesource.com/chromium/src/tools/grit/+/c1a76d6143016e9b2c292236bab623d1497cd31f/grit/format/data_pack.py # version 5
+seq:
+  - id: version
+    type: u4
+    doc: Support versions are 4 and 5
+  - id: header
+    type:
+      switch-on: version
+      cases:
+        4: header4
+        5: header5
+types:
+  header4:
+    seq:
+      - id: number_of_resources
+        type: u4
+      - id: encoding
+        type: u1
+        enum: encoding
+      - id: resources
+        type: resource
+        repeat: expr
+        repeat-expr: number_of_resources
+      - id: end_of_resource
+        type: end_of_resource
+        doc: |
+          The length is calculated by looking at the offset of
+          the next item, so an extra entry is stored with id 0
+          and offset pointing to the end of the resources.
+  header5:
+    seq:
+      - id: encoding
+        type: u1
+        enum: encoding
+      - id: ignore
+        size: 3
+        doc: These three bytes have no meaning in the file
+      - id: number_of_resources
+        type: u2
+      - id: number_of_aliases
+        type: u2
+      - id: resources
```

Comment:
Duplication

Reply:
What is duplicated?

Reply:
`number_of_resources` `resources` `end_of_resource` in both types. It is not how this is supposed to be specified. Instead there should be a single type, with `if`s conditioned on version.

Reply:
I don't really see the issue. I want to remove 'end_of_resource' anyway (as discussed) and then it is really just two very simple headers that have the various elements in a different order. Using "ifs" everywhere doesn't really help with readability either. Abstraction is good, but there is a thing as too much abstraction.

Reply:
I see the issue. Having 2 different formats for headers would cause unnecessary issues for strongly typed languages and even for non-strongly-typed ones and would be not enough clearly carrying the information how the format has been changed when the version has changed from 3 to 5.

Reply:
I was actually talking about it being more readable for humans for maintainability.